[Ervin Kin] iP

ErvinK123 opened the pull request:
# Puke 
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/)) 

Puke is a rude chatbot :  
- doesn't want to talk to you 
- text-based 
- ~~Easy~~ SUPER EASY to use 

Some features :
1. Add tasks
2. Delete tasks 
3. Check what's coming up  

Download it [here](https://github.com/nus-cs2103-AY2223S1/ip/pull/238) for **FREE**! 👍 

Should you download it? 
- [x] yes 
- [ ] no 

## 

If you Java programmer, you can use it to practice Java too. Here's the main method:

```java 
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ErvinK123 committed:
Level 1 echo functionality of puke chatbot

ErvinK123 committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

ErvinK123 committed:
Add the ability to mark tasks as done. Add the ability to change the status back to not done.

ErvinK123 committed:
Add support for tracking three types of tasks: Todo, Deadline, Events

ErvinK123 committed:
Preliminary handle the case for unrecognised commands level-4

ErvinK123 committed:
Added in exception handling for level-5

ErvinK123 committed:
Changed formatting of error messages to fit with chat bot

ErvinK123 committed:
Added delete functionality

ErvinK123 committed:
update testing

ErvinK123 committed:
Testing Scripts update

ErvinK123 committed:
Added date formating ability to dealines and events

ErvinK123 committed:
Remove unnecessary imports

ErvinK123 committed:
Add Save to hard disk functionality

ErvinK123 committed:
Merge branch 'branch-Level-8'

ErvinK123 committed:
Add Object Oriented Programming classes

ErvinK123 committed:
Organize into packages and update runtest.bat

ErvinK123 committed:
Merge branch 'add-gradle-support'

ErvinK123 committed:
Able to run duke using gradle

ErvinK123 committed:
Add JUnit testing

ErvinK123 committed:
Add JavaDoc

ErvinK123 committed:
Update Javadoc

ErvinK123 committed:
Add find functionality

ErvinK123 committed:
Tweak according to convention

ErvinK123 committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	pukeData.txt

ErvinK123 committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/puke/TaskList.java
Keep relevant fields, remove one duplicate field

ErvinK123 committed:
Add tutorial follow-along jfx

ErvinK123 committed:
Add GUI for puke
Test Features , runs from Duke instead of Main for now

ErvinK123 committed:
Run from Main.java
Remove unnecessary statement
Remove import and text UI

ErvinK123 committed:
Change Sizing

Marcusgwj commented on src/main/java/puke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+package puke;
```
I think you accidentally pushed your Deadline.java.swp file, would be good to delete it.

Marcusgwj commented on src/main/java/puke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    /**
+     * Get string representation of Deadline
```
Should Get should be changed to Gets?

Marcusgwj commented on src/main/java/puke/Duke.java:
```diff
@@ -0,0 +1,129 @@
+        }
+    }
+
+    private static boolean ended = false;
```
Perhaps can change ended naming to hasEnded

Marcusgwj commented on src/main/java/puke/Ui.java:
```diff
@@ -0,0 +1,98 @@
+    public void systemMessage(int i , Duke d, Task t) {
+        if (i == 1) {
+            //Bye message
+            System.out.println("    ____________________________________________________________");
```
Perhaps you can combine the strings into one single string and print once instead of printing multiple times. I noticed the same issue in several other places too.

Marcusgwj commented on src/main/java/puke/Task.java:
```diff
@@ -0,0 +1,54 @@
+/**
+ * Task - can take 3 forms , Deadline , Event or ToDo
+ */
+public abstract class Task {
```
Perhaps Task, Deadline, Event, ToDo can be placed in their own Task Package

Marcusgwj reviewed (commented):
LGTM, some nits to fix!

liviamil commented on src/main/java/puke/DialogBox.java:
```diff
@@ -0,0 +1,52 @@
+package puke;
```
Some missing javadocs for this class?

liviamil commented on src/main/java/puke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+
+    }
+
+    public static String getFindTask(String s) {
```
perhaps could use getTask instead of getFindTask, or include a javadoc, to avoid confusion

liviamil commented on src/main/java/puke/TaskList.java:
```diff
@@ -0,0 +1,73 @@
+    protected int numTasks = tasks.size();
+    protected int removed = 0;
+
+    /**
+     * how many tasks in the list
+     */
```
seems like a misplaced documentation?

liviamil commented on src/main/java/puke/ToDo.java:
```diff
@@ -0,0 +1,37 @@
+        super(description);
+    }
+
+    /**
+     * Get string representation of ToDo
+     * @return string representation of ToDo
+     */
```
Perhaps could use Gets?

liviamil reviewed (commented):
Looks good! Just commented on very minor things (a few missing javadocs, awkward spacings)

ErvinK123 committed:
Update build.gradle with JavaFX

The previous build.gradle file did not have support for JavaFX apps.
Changed main class name as well as added javafx support.

ErvinK123 committed:
Add time support to tasks

Only accepts Date parameters when creating tasks.

The tasks need a time field to be more accurate.

Let's change LocalDate to LocalDateTime to add time support

This allows Time field to be stored, accepted and displayed.

ErvinK123 committed:
Add assertions to puke() method

There are several important assumptions that must hold when running
the function.

Let's add assertions to ensure that they hold.

ErvinK123 committed:
Refactor Coding Style

Added missing JavaDocs and fixed indentation errors. Also removed
unnecessary import statements and parameters from methods.

ErvinK123 committed:
Change JavaFX version in fxml file

Warning given that Javafx 18 API was used to load, switched to
JavaFX 11.

Changed display format for empty task description.

ErvinK123 committed:
Add more assertions

Added assertions to MainWindow

ErvinK123 committed:
Remove unecessary functions

Removed some functions that were for text based GUI.

ErvinK123 committed:
Merge pull request #1 from ErvinK123/A-CodeQuality

Code Quality improvements

ErvinK123 committed:
Merge branch 'master' into heads/A-Assertions

ErvinK123 committed:
Merge pull request #2 from ErvinK123/heads/A-Assertions

Assertion

ErvinK123 committed:
Add help menu and intro

Help menu autoloaded on launch, added help command functionality

ErvinK123 committed:
Remove unnecessary import statement

ErvinK123 committed:
Add Exception handling
- No element found
- Index out of bounds

Add credit

ErvinK123 committed:
Update help menu

Refactor code quality

Add icon and name to javafx app

ErvinK123 committed:
Fix bye functionality

Now exits and closes the app when bye command is given

ErvinK123 committed:
Add User Guide

Updated README with User Guide for Puke bot.

ErvinK123 committed:
Add UI picture for github Pages

ErvinK123 committed:
Rename UI picture according to format specified

ErvinK123 committed:
Rename Ui picture for IP showcase

Second commit as first had issue

ErvinK123 committed:
Resolve naming issue with IP showcase picture

ErvinK123 committed:
Add support for MACOS storage

Remove unecessary import

ErvinK123 committed:
Fix text overflow bug, UI breakages

ErvinK123 committed:
Update README
